src/lucky/tags/text_helpers.cr

```diff
@@ -0,0 +1,244 @@
+    if text
+      content = truncate_text(text, length, omission, separator)
+      content = escape ? HTML.escape(content) : content
+      content += blk.call.to_s if !blk.nil? && text.size > length
+      content
+    end
+  end
+
+  def truncate(text : String, length : Int32 = 30, omission : String = "...", separator : String | Nil = nil, escape : Bool = true, &block : -> _)
+    truncate(text, length, omission, separator, escape, blk: block)
+  end
+
+  private def truncate_text(text : String, truncate_at : Int32, omission : String = "...", separator : String | Nil = nil)
+    return text unless text.size > truncate_at
+
+    length_with_room_for_omission = truncate_at - omission.size
+    stop = \
+      if separator
+        text.rindex(separator, length_with_room_for_omission) || length_with_room_for_omission
+      else
+        length_with_room_for_omission
+      end
+
+    "#{text[0, stop]}#{omission}"
+  end
+
+  def highlight(text : String, phrases : Array(String | Regex), highlighter : Proc | String = "<mark>\\1</mark>")
+    if text.blank? || phrases.all?(&.to_s.blank?)
+      text || ""
+    else
+      match = phrases.map do |p|
+        p.is_a?(Regex) ? p.to_s : Regex.escape(p.to_s)
+      end.join("|")
+
+      if highlighter.is_a?(Proc)
+        text.gsub(/(#{match})(?![^<]*?>)/i, &highlighter)
+      else
+        text.gsub(/(#{match})(?![^<]*?>)/i, highlighter)
+      end
+    end
+  end
+
+  def highlight(text : String, phrases : Array(String | Regex), &block : String -> _)
+    highlight(text, phrases, highlighter: block)
+  end
+
+  def highlight(text : String, phrase : String | Regex, highlighter : Proc | String = "<mark>\\1</mark>")
+    phrases = [phrase] of String | Regex
+    highlight(text, phrases, highlighter)
+  end
+
+  def highlight(text : String, phrase : String | Regex, &block : String -> _)
+    phrases = [phrase] of String | Regex
+    highlight(text, phrases, highlighter: block)
+  end
+
+  def excerpt(text : String, phrase : Regex | String, separator : String = "", radius : Int32 = 100, omission : String = "...")
+    return "" if text.to_s.blank?
+
+    case phrase
+    when Regex
+      regex = phrase
+    else
+      regex = /#{Regex.escape(phrase.to_s)}/i
+    end
+
+    return unless matches = text.match(regex)
+    phrase = matches[0]
+
+    unless separator.empty?
+      text.split(separator).each do |value|
+        if value.match(regex)
+          phrase = value
+          break
+        end
+      end
+    end
+
+    first_part, second_part = text.split(phrase, 2)
+
+    prefix, first_part   = cut_excerpt_part(:first, first_part, separator, radius, omission)
+    postfix, second_part = cut_excerpt_part(:second, second_part, separator, radius, omission)
+
+    affix = [first_part, separator, phrase, separator, second_part].join.strip
+    [prefix, affix, postfix].join
+  end
+
+  def pluralize(count : Int32 | String | Nil, singular : String, plural = nil)
```

Comment:
Sorry I didn't see this earlier. I think that the count should only be `Int32`. If people want to use a string they can call `to_i` on it. I think this is important because most of the time you want to use an Int32, and if you accidentally pass String, you'd want to know. If you really did mean to pass a String, calling `to_i` in those rare instances should be ok

Reply:
It's a feature in it's current form not an accident enabler, I deliberated on this one a bit when adding the types. The spec shows some of the flexibility you get in it's current form.

```crystal
view.pluralize("1,066", "count").should eq "1,066 counts"
view.pluralize("1.25", "count").should eq "1.25 counts"
view.pluralize("1.0", "count").should eq "1.0 count"
view.pluralize("1.00", "count").should eq "1.00 count"
view.pluralize(2, "count", "counters").should eq "2 counters"
view.pluralize(nil, "count", "counters").should eq "0 counters"
```

Reply:
I think we could cover most of those cases by also allowing `Float`. Then people could call `to_f` on the string. I'm not totally sure about the `nil` case, but I'd rather remove it because it seems odd to have something be `nil` like that in practice. Usually you would use an array of items and call `size`, so you shouldn't have `nil` in most cases. I'd rather be a bit more strict early on and then add more types. That way we know for sure that people need them and know the exact use-case of it

Reply:
Actually you're right about keeping strings. I can see why that would be useful so you can pass a formatted/localized number. In the future we may want to try to do it with a block though so that you pass in an int or float and can format it in a block or something: 

```crystal
pluralize(1_300, "thing") do |number|
  humanize_number(number) # returns "1,300"
end
```

If that's not too much work we could do it in this PR, but might be better to do separately so it doesn't hold this one up too long. I still think we should remove the `nil` option though. I could see that causing bugs